### ISSUE TITLE
App not installed error for User Company Communicator App (users are not recieving messages)

### ISSUE THREAD

**Issue by ArunkumarTMEYGDS** [OfficeDev/microsoft-teams-apps-company-communicator#1575]
Hi Team,

We have been able to deploy the company communicator app.

As a part of testing we tried sending message to Team, where a test users were part of.
We are able to draft and preview the message in Author app. While trying to send it to the users via Team (M365 Group) we are getting error with no users recieving messages. On checking the report logs, it mentions App not installed for all the users, where as as logged in as test user, we can see the App present and can open to check for messages. 

Sharing screenshot of Log report:
<img width="960" alt="Error" src="https://github.com/user-attachments/assets/a43c71b1-4239-4390-9d6d-515bcecc5638">

Please advice what could be the reason. We have tried to the following troubleshooting steps:
1. Tried removing and adding app both user and author with new version number.
2. Tried syncing and restarting function apps.

**Comment by peddivyshnavi** [comment]
Hi @ArunkumarTMEYGDS 
Thanks for reaching out to us 
we will check and will get back to you within 48 Hours.

**Comment by peddivyshnavi** [comment]
Hi @ArunkumarTMEYGDS 
we suspect this issue is related to app policies.
As a part of troubleshooting call which we have attended for GitHub issue([1546](https://github.com/OfficeDev/microsoft-teams-apps-company-communicator/issues/1546)) we see team's policies are not assigned to Global (Org-wide default) they were assigned to custom policies.
could you please reassign it to Global (Org-wide default) 
<img width="834" alt="image" src="https://github.com/user-attachments/assets/e9ed1b00-de58-4b88-91d2-fc960103028f">

**Comment by ArunkumarTMEYGDS** [comment]
Hi @peddivyshnavi,

Sharing details from our Teams Platform Team below:

1. Our Dev and QA Teams tenant are both migrated to group based management so app permission polices no longer apply.
2. The apps are now assigned to users or groups on an app by app basis, Tech News user and Admin apps are currently assigned to Everyone. 

Can these be causing any issue in terms of sending the message?

**Comment by peddivyshnavi** [comment]
Hi @ArunkumarTMEYGDS 
Could you please confirm whether you were able to send messages through remaining options.

**Comment by ArunkumarTMEYGDS** [comment]
@peddivyshnavi No we are not able to send via any options.

**Comment by peddivyshnavi** [comment]
Hi @ArunkumarTMEYGDS 
we were checking on the issue and will get back to you within 48 Hours.

**Comment by peddivyshnavi** [comment]
Hi @ArunkumarTMEYGDS 
we are checking on this issue will update you as soon as possible

**Comment by clarkec2** [comment]
So investigating this we noticed that all the users who we see with the "-2 : App Not Installed" error are missing from the storage account tables entirely. We have broadened our tests and it appears to be affecting around 10% of the users in our org.  I can manually add them to use the tables and they are able to receive messages, however with it affecting 10% of the user base that works out to potentially 5k users. What would the best method be to automatically push the app out again? Would reuploading and assigning it to the policy be best or does it need to be redeployed for the proactive install to kick off?

**Comment by peddivyshnavi** [comment]
@ArunkumarTMEYGDS 
We are trying to migrate to app centric for analyzing the issue

**Comment by Saraanuzzi** [comment]
We are also facing this problem in our organization, I have already carried out some tests such as removing and including the user in the database, I have restarted the application and without success.

**Comment by ArunkumarTMEYGDS** [comment]
Hi @peddivyshnavi 

Is there any resolution to this? Do you have any updates?

**Comment by peddivyshnavi** [comment]
We apologize for the delay in our response and will get back to you as soon as possible.

**Comment by peddivyshnavi** [comment]
Hi @ArunkumarTMEYGDS 
We were checking this issue internally we will let you know ASAP

**Comment by peddivyshnavi** [comment]
Hi @ArunkumarTMEYGDS 
thanks for your patience
We are currently working to find a solution for this issue, as it is new to us. We are attempting to reproduce the problem and will get back to you as soon as possible.

**Comment by peddivyshnavi** [comment]
Hi @ArunkumarTMEYGDS
thanks for your patience
We are currently working to find a solution for this issue, as it is new to us. We are attempting to reproduce the problem and will get back to you as soon as possible.

**Comment by ArunkumarTMEYGDS** [comment]
Hi @peddivyshnavi 

Please advice. We are still facing the issue with few users still not recieving messages, error being thrown - app not installed. A A test replica of the app is able send messages to same user. This is really a road blocker for us.

**Comment by peddivyshnavi** [comment]
Hi @ArunkumarTMEYGDS 
could you please confirm us whether test replica is implemented in app centric policy
and Is the test replica sending messages to users who previously faced the "app not installed" error?

**Comment by ArunkumarTMEYGDS** [comment]
Yes @peddivyshnavi, they are in app centric policy and assigned to global org wide default. The test replica is able to send message to user but not the other one.

**Comment by ArunkumarTMEYGDS** [comment]
Yes @peddivyshnavi, they are in app centric policy and assigned to global org wide default. The test replica is able to send message to user but not the other one.

**Comment by ArunkumarTMEYGDS** [comment]
Hi @peddivyshnavi 
 
Can we get into a quick call. We need the app to send message next week globally and still we are facing issue with message sent to partial users .

**Comment by ArunkumarTMEYGDS** [comment]
Hi @peddivyshnavi,

**We tried the below troubleshotting steps shared with us but still we are getting same error.**

Remove user IDs from the userdata table for users with the "app not installed" error.
Please follow below steps:
Please export the status of the delivered message; you will receive a delivery table.
Using Delivery Status, we can identify users whose messages are not being delivered. Please copy the user IDs from the table for those affected.
Please follow the steps below to delete the impacted users record from the UserData table.
1.    Login to Azure portal
2.    Go to the resource group where CC is deployed.
3.    Go to Storage account.
4.    Click on Storage browser.
5.    Click on Tables.
6.    Click on UserData table.
7.    Click on Advance Filter option
8.    Paste and Run the query which will be in below format  - Sample Query for same is mentioned below - 
PartitionKey eq 'UserData' and RowKey eq 'UserId' or RowKey eq 'UserId'
9.    Select and Delete all the data rows from the query result.
10.    Try to resend the message to the users again.

**Comment by clarkec2** [comment]
I keep getting emails on this issue so I am going to chime in because we were able to fix it. 

After cleaning up the tables, go into TAC and follow these steps for whatever policies you have it setup on
https://learn.microsoft.com/en-us/microsoftteams/install-teams-apps#install-apps-using-app-setup-policy

or if youre on the app centric management thingy...
https://learn.microsoft.com/en-us/microsoftteams/install-teams-apps#install-apps-using-app-centric-management

Alternately this command will work too...

Set-CsTeamsAppSetupPolicy -Identity 'SETUP POLICY NAME' -AppPresetList "APP NAME"

**Comment by ArunkumarTMEYGDS** [comment]
@peddivyshnavi  Do we have an update on this issue. The above resolutions did not fix our issue. As per discussion the policies are set as intended. Please note this is a road blocker for us.

**Comment by ArunkumarTMEYGDS** [comment]
Hi @peddivyshnavi,

We tried the below steps as advised but still same issue.

Try disconnecting and reconnecting the function apps. For that please follow below steps

1. Open function app
2. Under deployment you will find option called deployment center
3. Please disconnect from repo its pointing to
4. And reconnect to official repo after 1hour
5. Please repeat the same for all function apps

**Comment by ArunkumarTMEYGDS** [comment]
Hi @peddivyshnavi,

We tried the below steps as advised but still same issue.

Try disconnecting and reconnecting the function apps. For that please follow below steps

1. Open function app
2. Under deployment you will find option called deployment center
3. Please disconnect from repo its pointing to
4. And reconnect to official repo after 1hour
5. Please repeat the same for all function apps

**Comment by ArunkumarTMEYGDS** [comment]
Hi @peddivyshnavi 

Is there any updates on this issue?

**Comment by tiwariShiv7** [comment]
Hi @ArunkumarTMEYGDS, No update currently. Till now We have tried to recreate the issue our side but it is not getting replicated. We are continuously making changes to our deployed solutions to analyze this issue

**Comment by tiwariShiv7** [comment]
Hi @ArunkumarTMEYGDS, We are still looking into it. Currently the issue is not getting recreated at our end. We are trying different configuration to reproduce it for analysis the possible causes.

**Comment by ArunkumarTMEYGDS** [comment]
Thanks @tiwariShiv7 . Awaiting further updates on the issue.

**Comment by ArunkumarTMEYGDS** [comment]
Hi @tiwariShiv7, 

Any progress on the issue. Also sharing the results of recent notification triggers
We have noticed initially the failure rate was ~25% but then in one case it was 43%.

Mostly the errors were App not installed and in several cases the errors were “429 : Failed” and “403 : Failed”.
 
<img width="711" alt="111" src="https://github.com/user-attachments/assets/56442ab1-9ac6-49d8-b2c4-84457d5d8b04">

**Comment by MirianaRebeggianiSacmi** [comment]
We have the same issue:
![image](https://github.com/user-attachments/assets/fc98bdcf-ec2b-4eb9-a104-246eded9e8db)
![image](https://github.com/user-attachments/assets/cabaab79-f439-47cd-937b-ae27a4cda9f7)

**Comment by v-jaygupta** [comment]
@ArunkumarTMEYGDS ,
Please let us know your availability for a call to troubleshoot the issue for 19th or 20th Nov.

**Comment by v-jaygupta** [comment]
@ArunkumarTMEYGDS ,
We have shared a meeting invite for the above suggested time.

**Comment by MirianaRebeggianiSacmi** [comment]
Hi, can we have a call for  Thursday 21th?  @v-jaygupta

**Comment by ArunkumarTMEYGDS** [comment]
Hi @v-jaygupta Do you have any update on this issue?

**Comment by ArunkumarTMEYGDS** [comment]
Hi Team, Please share an update on this issue. The issue has been unresolved for over a span of 2 months now.

**Comment by v-jaygupta** [comment]
@ArunkumarTMEYGDS ,

Thank you for your patience regarding this issue!
Following internal discussions, the team has recommended enabling the proactive installation feature in the App service to streamline the installation of the user app from the Company Communicator's side. This will allow the app to retrieve user information and send messages accordingly. 
Additionally, we have attempted to replicate the scenario across various deployments and demo tenants on our end, and it appears to be functioning as expected.

**Comment by ArunkumarTMEYGDS** [comment]
HI All, we may not be able to enable proactive installation feature in the App service. We are still looking into this if it can be done.

**Comment by LokCodeSample** [comment]
Hi Team, 
We updated our MS Teams environment to support Group based policy and tested further but some users are still not receiving messages in Company Communicator user app. Can you please reopen the ticket and continue troubleshoot the issue further.

**Comment by kunniyoorSreejitEY** [comment]
We updated our MS Teams environment to support Group based policy and tested further but some users are still not receiving messages in Company Communicator user app. Can you please reopen the ticket and continue troubleshoot the issue further.

**Comment by ArunkumarTMEYGDS** [comment]
Hi Team,

As mentioned by LokCodeSample and kunniyoorSreejitEY, the proposed fix did not work and we are still seeing same users are marked as App not installed and message not sent across. Please help us in mitigating this issue. It has been ongoing for a very long time now.

**Comment by v-jaygupta** [comment]
Thanks for the update, we will check on this and update you with the next steps soon.

**Comment by v-milibiswas** [comment]
Hello @ArunkumarTMEYGDS ,
We would like to inform you that the Company Communicator App Template will officially be retired as of March 31, 2025. While we will no longer provide issue monitoring or implement feature enhancements for this app, you are welcome to continue using the existing codebase in its current form.
As an alternative, we highly recommend exploring Viva Amplify—a comprehensive and feature-rich solution tailored to address your communication needs with enhanced functionality and ongoing support.
Thank you for your understanding and for your valued engagement with the Company Communicator App throughout its journey.
